src/app/mork/models/users.py

```diff
@@ -0,0 +1,94 @@
+"""Mork users models."""
+
+import enum
+from typing import List
+from uuid import uuid4
+
+from sqlalchemy import (
+    Enum,
+    ForeignKey,
+    Index,
+    Integer,
+    String,
+    UniqueConstraint,
+)
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from mork.models import Base
+
+
+class ServiceName(str, enum.Enum):
+    """Enum for service names."""
+
+    ASHLEY = "ashley"
+    EDX = "edx"
+    BREVO = "brevo"
+    JOANIE = "joanie"
+
+
+class DeletionReason(str, enum.Enum):
+    """Enum for the reason of the deletion request."""
+
+    USER_REQUESTED = "user_requested"
+    GDPR = "gdpr"
+
+
+class DeletionStatus(str, enum.Enum):
+    """Enum for deletion statuses."""
+
+    TO_DELETE = "to_delete"
+    DELETED = "deleted"
+
+
+class ServiceStatus(Base):
+    """Table for storing the user status for a service."""
+
+    __tablename__ = "service_statuses"
+    __table_args__ = (
+        UniqueConstraint("user_id", "service_name", name="uq_record_service"),
+        Index(
+            "idx_user_service_status",
+            "user_id",
+            "service_name",
+            "status",
+        ),
+        Index("idx_service_status", "service_name", "status"),
+    )
+
+    id: Mapped[UUID] = mapped_column(
+        UUID(as_uuid=True), primary_key=True, default=uuid4
+    )
+    user_id: Mapped[UUID] = mapped_column(
+        ForeignKey("users.id", ondelete="CASCADE"), nullable=False
+    )
+    service_name: Mapped[ServiceName] = mapped_column(String(10), nullable=False)
```

Comment:
Similar to `status` typing, you can type with `ServiceName` enum values
```suggestion
    service_name: Mapped[ServiceName] = mapped_column(Enum(ServiceName), nullable=False)

```